api/filters/namespace/namespace.go

```diff
@@ -138,34 +127,52 @@ func (ns Filter) roleBindingHack(obj *yaml.RNode, gvk resid.Gvk) error {
 			return err
 		}
 
-		return ns.trackableSetter.SetEntry("", ns.Namespace, yaml.NodeTagString)(node)
+		return ns.fieldSetter()(node)
 	})
 
 	return err
 }
 
-// removeFieldSpecsForHacks removes from the list fieldspecs that
+func isRoleBinding(kind string) bool {
+	switch kind {
+	case roleBindingKind, clusterRoleBindingKind:
+		return true
+	default:
+		return false
```

Comment:
the switch statement seems a bit overkill for this check, couldn't we simply `return gvk.Kind == roleBindingKind || gvk.Kind == clusterRoleBindingKind`?